File: cpp/3/25304.cpp
#include <iostream>

using namespace std;

int main() {
    int result = 0;
    int total, amount;
    cin >> total >> amount;

    for (int i = 0; i < amount; i++) {
        int x, y;
        cin >> x >> y;
        result += x * y;
    }

    if (total == result)
        cout << "Yes";
    else
        cout << "No";

    return 0;
}

File: kotlin/Level/3/25304.kt
fun main() {
	var total = readLine()!!.toInt()
    var amount = readLine()!!.toInt()
    var result = 0

    for (i: Int in 1..amount) {
        var x = readLine()!!.split(" ").map { it.toInt() }
        result += x[0] * x[1]
    }

    if (total == result)
        print("Yes")
    else
        print("No")
}